test/powershell/engine/Remoting/SSHRemotingAPI.Tests.ps1

```diff
@@ -1,41 +1,68 @@
+Import-Module $PSScriptRoot\..\..\Common\Test.Helpers.psm1
+
 Describe "SSH Remoting API Tests" -Tags "Feature" {
 
     Context "SSHConnectionInfo Class Tests" {
 
         It "SSHConnectionInfo constructor should throw null argument exception for null HostName parameter" {
 
+            { [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
+                "UserName",
+                [System.Management.Automation.Internal.AutomationNull]::Value,
+                [System.Management.Automation.Internal.AutomationNull]::Value,
+                0) } | ShouldBeErrorId "PSArgumentNullException"
+        }
+
+        It "SSHConnectionInfo should throw file not found exception for invalid key file path" {
+
             try
             {
-                [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
+                $sshConnectionInfo = [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
                     "UserName",
-                    [System.Management.Automation.Internal.AutomationNull]::Value,
-                    [System.Management.Automation.Internal.AutomationNull]::Value)
+                    "localhost",
+                    "NoValidKeyFilePath",
+                    22)
 
+                $rs = [runspacefactory]::CreateRunspace($sshConnectionInfo)
+                $rs.Open()
+                
                 throw "No Exception!"
             }
             catch
             {
-                $_.FullyQualifiedErrorId | Should Be "PSArgumentNullException"
+                $expectedFileNotFoundException = $_.Exception
+                if ($_.Exception.InnerException -ne $null -and $_.Exception.InnerException.InnerException -ne $null)
+                {
+                    $expectedFileNotFoundException = $_.Exception.InnerException.InnerException
+                }
+                
+                $expectedFileNotFoundException | Should BeOfType "System.IO.FileNotFoundException"
```

Comment:
It would appear that ShouldBeErrorId doesn't return the thrown exception, so we would be unable to process $exc further here (ShouldBeErrorId will return result of Should Be - bool).

Based on this, would you prefer for ShouldBeErrorId to be implemented here to check for ``PSRemotingDataStructureException``, or continue to inspect the nested inner exception?